plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/PrivilegeComponent.groovy

```diff
@@ -104,20 +106,22 @@ extends DirectComponentSupport
     }
     def manager = securitySystem.getAuthorizationManager(DEFAULT_SOURCE)
     List<PrivilegeXO> created = []
-    ['create', 'read', 'update', 'delete'].each { method ->
-      created << manager.addPrivilege(
-          new Privilege(
-              id: Long.toHexString(System.nanoTime()),
-              name: privilegeXO.name ? "${privilegeXO.name} - (${method})" : null,
-              description: privilegeXO.description,
-              type: TargetPrivilegeDescriptor.TYPE,
-              readOnly: false,
-              properties: [
-                  (ApplicationPrivilegeMethodPropertyDescriptor.ID): method,
-                  (TargetPrivilegeRepositoryTargetPropertyDescriptor.ID): privilegeXO.repositoryTargetId,
-                  (TargetPrivilegeRepositoryPropertyDescriptor.ID): repositoryId,
-                  (TargetPrivilegeGroupPropertyDescriptor.ID): groupId
-              ]
+    ['create', 'read', 'update', 'delete'].each { String method ->
+      created << asPrivilegeXO(
```

Comment:
Ran into a real typing issue here where we filled the List<PrivilegeXO> with Privilege objects and had it work mostly due to the fact that they are essentially copies of each other. Should be fixed regardless of decisions made regarding this PR.